abdm_python_integrator/utils.py

```diff
@@ -0,0 +1,78 @@
+import functools
+import json
+
+import requests
+
+from abdm_python_integrator.const import SESSIONS_PATH
+from abdm_python_integrator.exceptions import ABDMAccessTokenException, ABDMConfigurationError
+from abdm_python_integrator.settings import app_settings
+
+
+class ABDMRequestHelper:
+    REQUIRED_CONFIGS = ['ABHA_URL', 'GATEWAY_URL', 'CLIENT_ID', 'CLIENT_SECRET', 'X_CM_ID']
+    gateway_base_url = None
+    abha_base_url = None
+    token_payload = None
+
+    @classmethod
+    @functools.cache
+    def _check_configs(cls):
+        if any(not hasattr(app_settings, config) for config in cls.REQUIRED_CONFIGS):
```

Comment:
Is there a better place to do validation of the settings? (like when the app is initialized or some other place). May be https://docs.djangoproject.com/en/4.2/topics/checks/ or some other place?

Reply:
Thanks for catching this. This check is no longer useful after implementing [AppSettings](https://github.com/dimagi/abdm-python-integrator/blob/ay/milestone-1/abdm_python_integrator/settings.py#L22) which sets default value to None.
I should change it to check if value is not None instead. 

The reason I added here (as runtime while execution ) is to avoid raising Exception in HQ environments where this integration will not be available.  Let me know if it makes sense .

Reply:
> The reason I added here (as runtime while execution ) is to avoid raising Exception in HQ environments where this integration will not be available

This code is active when a URL is hit. The urls should not even be added where the integration is not available, so it shouldn't even come here for the exception to be raised, right?

Reply:
Yes that makes sense. In that case, this check does not seem particularly useful and could be removed ?

Along with urls , I think this app should not be included in installed apps (_has migrations_). I am going to create specs for settings part (hq and cloud) and share for review.

Reply:
@sravfeyn Thinking more on this, yes I agree these checks if to be included could be done at a better place like one you suggested. 
I am thinking of adding these settings validation check as a part of last PR once I am done including a couple more settings if that is good by you.

Reply:
Removed here : 5a1c2996e5a9da64ffddb5f9488398121243f61a
Ticket (just for keeping track): https://dimagi-dev.atlassian.net/browse/INDIV-223